src/transformers/generation_tf_utils.py

```diff
@@ -871,7 +877,12 @@ def _generate_no_beam_search(
 
         while cur_len < max_length:
             model_inputs = self.prepare_inputs_for_generation(
-                input_ids, past=past, attention_mask=attention_mask, use_cache=use_cache, **kwargs
+                input_ids,
+                past=past,
+                attention_mask=attention_mask,
+                use_cache=use_cache,
+                encoder_outputs=encoder_outputs,
```

Comment:
If past and encoder_outputs is the same, why do we pass both here? Because the name `enocder_outputs` is cleaner? Would it maybe be better to keep `past` for now for consistency?

Reply:
Once again, your comment led to insightful changes -- digging deeper, I found that the `past_key_values` output of the decoder is different on PT and TF (TF's is a tuple with the encoder's hidden states and past key values, and PT's is only the later). I had adapted the code from PT, so my `past_key_values` was missing the encoder outputs, and I had to pass them explicitly in order to work.

I've reverted this change and made TFSpeech2Text's `past_key_values` match the expected TF shape. An updated TF generate would probably have prevented this error 🤔